client/src/partials/caution/caution.js

```diff
@@ -251,6 +251,35 @@ angular.module('bhima.controllers')
       $scope.setCurrency(sessionDefault);
     }
 
+    function haltOnNoExchange () {
+      if (exchange.hasDailyRate()) { return; }
+
+      var instance = $modal.open({
+        templateUrl : 'noExchangeRate.html',
+        backdrop    : 'static',
+        keyboard    : false,
+        controller  : function ($scope, $modalInstance) {
```

Comment:
You could make an `exchangeModalController` somewhere else and then use it here.  For example:
```js
angular.module('bhima.controllers').controller('ExchangeModalController', ['$scope', '$modalInstance', ExchangeModalController]);

function ExchangeModalController($scope, $modalInstance) {
  $scope.timestamp= new Date();
  $scope.close = $modalInstance.dismiss;
  $scope.setExchange = $modalInstance.close;
}
```